plugin/src/main/java/io/jenkins/plugins/analysis/warnings/axivion/AxivionSuite.java

```diff
@@ -139,6 +139,12 @@ public void setNamedFilter(final String namedFilter) {
     public Report scan(final Run<?, ?> run, final FilePath workspace, final Charset sourceCodeEncoding,
             final LogHandler logger) throws ParsingException, ParsingCanceledException {
 
+        // field was added in 9.1.0
```

Comment:
Please move that code to a `readResolve` method. It would be also helpful in such cases to have a test that tries to work on an instance in the old format. 

Example Serialization: 
https://github.com/jenkinsci/warnings-ng-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/ResultAction.java#L105

Example Test:
https://github.com/jenkinsci/warnings-ng-plugin/blob/master/plugin/src/test/java/io/jenkins/plugins/analysis/core/model/ReportXmlStreamTest.java

Reply:
:+1: thanks for pointing me to the classes; I've added a testcase loading a 9.0.1 version of the tool.